<a href="https://colab.research.google.com/github/EdgarVitelaChicharo/Dise-o-y-Anal-sis-de-Algoritmos---FES/blob/main/Estrategias_Dise%C3%B1o.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Fuerza Bruta


Ejemplo 1, busqueda de patrones de texto.

In [ ]:
def string_match_brute( A , P ):
    for i in range( len(A) - len(P) + 1   ):
        for j in range( len(P) ):
            if P[j] == A[ i+j ]:
                pass
            else:
                break
        if j+1 == len(P) and A[i+j] == P[ j ]:
            print(f"Se encontró en la posición:{ i }")

entrada = "001110010101101110101101"
patron = "01101"
string_match_brute(entrada , patron)
              

Se encontró en la posición:10
Se encontró en la posición:19


In [87]:
# Inicio del método
def string_match_brute( A , P ):
    
    cantidad = 0
    indices = []

    for i in range(len(entrada)):
        if palabra == entrada[i]:
            cantidad += 1
            indices.append(i)

    print(f"{palabra} se repite {cantidad} veces")
    print(f"{palabra}  esta en los indices {indices}")
# Fin del método

archivo = open('/content/Quijote.txt')
texto = archivo.read()
quitar = ",;:./¿?¡!<>{}[]'()-–_‘»"

for caracter in quitar:
    texto = texto.replace(caracter,"")

texto = texto.lower()
entrada = texto.split(" ")

palabra = "don"

string_match_brute(entrada , palabra)

don se repite 1585 veces
don  esta en los indices [0, 21, 40, 363, 444, 538, 603, 640, 681, 729, 763, 770, 863, 925, 943, 1056, 1148, 1211, 2335, 2810, 2839, 2990, 3023, 3065, 3073, 3127, 3169, 3351, 3366, 3544, 3562, 3708, 3866, 3897, 4024, 4037, 4064, 4247, 4297, 4479, 4587, 4605, 4716, 5016, 5067, 5124, 5206, 5293, 5352, 5445, 5500, 5567, 5590, 5629, 5693, 5704, 5962, 6025, 6038, 6122, 6214, 6358, 6388, 6586, 6650, 6665, 6843, 6883, 7010, 7021, 7101, 7395, 7410, 7581, 7639, 7860, 7973, 8003, 8124, 8434, 8824, 9005, 9027, 9076, 9103, 9327, 9685, 9862, 10008, 10206, 10245, 10284, 10508, 11654, 11818, 11850, 11980, 12636, 12651, 12694, 12831, 12929, 12979, 13453, 13784, 14587, 14669, 14694, 15043, 15185, 15230, 15298, 15322, 15346, 15398, 15462, 15577, 15738, 15778, 16264, 16360, 16416, 16910, 16968, 17087, 17107, 17139, 17168, 17203, 17232, 17274, 17300, 17362, 17500, 17672, 17729, 17851, 18310, 19114, 19150, 19227, 19374, 19460, 19555, 19690, 19741, 19762, 19800, 19823, 19885, 19948,

Cantidad de veces que se repiten las palabras#

don se repite 1585 veces.

hidalgo se repite 40 veces.

merced se repite 545 veces.

don  esta en los indices [0, 21, 40, 363, 444, 538, 603, 640, 681, 729, 763, 770, 863, 925, 943, 1056, 1148, 1211, 2335, 2810, 2839, 2990, 3023, 3065, 3073, 3127, 3169, 3351, 3366, 3544, 3562, 3708, 3866, 3897, 4024, 4037, 4064, 4247, 4297, 4479, 4587, 4605, 4716, 5016, 5067, 5124, 5206, 5293, 5352, 5445, 5500, 5567, 5590, 5629, 5693, 5704, 5962, 6025, 6038, 6122, 6214, 6358, 6388, 6586, 6650, 6665, 6843, 6883, 7010, 7021, 7101, 7395, 7410, 7581, 7639, 7860, 7973, 8003, 8124, 8434, 8824, 9005, 9027, 9076, 9103, 9327, 9685, 9862, 10008, 10206, 10245, 10284, 10508, 11654, 11818, 11850, 11980, 12636, 12651, 12694, 12831, 12929, 12979, 13453, 13784, 14587, 14669, 14694, 15043, 15185, 15230, 15298, 15322, 15346, 15398, 15462, 15577, 15738, 15778, 16264, 16360, 16416, 16910, 16968, 17087, 17107, 17139, 17168, 17203, 17232, 17274, 17300, 17362, 17500, 17672, 17729, 17851, 18310, 19114, 19150, 19227, 19374, 19460, 19555, 19690, 19741, 19762, 19800, 19823, 19885, 19948, 20018, 20110, 20221, 20295, 20395, 20486, 20556, 20630, 20700, 20822, 20879, 20994, 21146, 21253, 21336, 21432, 21525, 21860, 21894, 22055, 22575, 22671, 22682, 22728, 22782, 22939, 23030, 23100, 23157, 23171, 23299, 23320, 23505, 23745, 23772, 23929, 24259, 24325, 24412, 24527, 24542, 24728, 24794, 25083, 25231, 25293, 25402, 25416, 25656, 25776, 25812, 25902, 26058, 26086, 26098, 26177, 26201, 26311, 26384, 26448, 26456, 26623, 26750, 26873, 26894, 26912, 26973, 27044, 27217, 27340, 27452, 27976, 28097, 28153, 28220, 28289, 28551, 28660, 28707, 28757, 28860, 28868, 28918, 28952, 31374, 31391, 31779, 31814, 31871, 31874, 31950, 31986, 32117, 32131, 32294, 32313, 32373, 32464, 33578, 33722, 33731, 33782, 33833, 33844, 33878, 33929, 33969, 34119, 34165, 34190, 34218, 34297, 34390, 34417, 34639, 34683, 34732, 34903, 35097, 35109, 35210, 35309, 35325, 35380, 35442, 35478, 35509, 35547, 35598, 35637, 35737, 35779, 35860, 35986, 36080, 36088, 36107, 36366, 36485, 36887, 37035, 37066, 37071, 37118, 37178, 37405, 37495, 37692, 37724, 37767, 37775, 37797, 37857, 38217, 38220, 38273, 38528, 39490, 39563, 39567, 39657, 39671, 39826, 39920, 39961, 40061, 40274, 40292, 40362, 40432, 40517, 40580, 40648, 40706, 40796, 40840, 40922, 41048, 41071, 41203, 41286, 41409, 41459, 41561, 41579, 41722, 41849, 41920, 41938, 42085, 42180, 42265, 42286, 42299, 42379, 42457, 42494, 42586, 42604, 42620, 42750, 42755, 43237, 43280, 43283, 43383, 43386, 43412, 43416, 43430, 43446, 43452, 43554, 43576, 43594, 43604, 43647, 43671, 43689, 43746, 43888, 43936, 43943, 43952, 44001, 44071, 44076, 44090, 44093, 44096, 44133, 44180, 44206, 44219, 44347, 44375, 44397, 44411, 44686, 44724, 44740, 44747, 44777, 44931, 44962, 45013, 45038, 45070, 45075, 45136, 45236, 45239, 45267, 45286, 45508, 45514, 45530, 45639, 45645, 45674, 45684, 45785, 45792, 45827, 45834, 46021, 46042, 46085, 46088, 46165, 46188, 46290, 46342, 46372, 46377, 46493, 46511, 46585, 46624, 46695, 47169, 47294, 47571, 47900, 47993, 48415, 49032, 49107, 49128, 49133, 49249, 49267, 49341, 49380, 49451, 49925, 50050, 50327, 50656, 50749, 51171, 51788, 51863, 51910, 51936, 52148, 52217, 52427, 52532, 53133, 53210, 53350, 53981, 54174, 54204, 54265, 54387, 54455, 54488, 54546, 54770, 54843, 54953, 54984, 55246, 55750, 55919, 56784, 56836, 57217, 57354, 57369, 57483, 57726, 58073, 58098, 58156, 58338, 58870, 58933, 58987, 59093, 59168, 59189, 59235, 59250, 59505, 59649, 59699, 59739, 59784, 59811, 59968, 60004, 60062, 60083, 60127, 60574, 60764, 60814, 60831, 61675, 62135, 62194, 62317, 62380, 62410, 62422, 62497, 62518, 62536, 62566, 62664, 62695, 62750, 63076, 63150, 63414, 63682, 63811, 63942, 63996, 64210, 64447, 64491, 64642, 64780, 64880, 64960, 64986, 65070, 65237, 65285, 65509, 65592, 66161, 66235, 66334, 66372, 67765, 67811, 68048, 68267, 68305, 68333, 68488, 68534, 68619, 68645, 68745, 69243, 69272, 69292, 69350, 69518, 69593, 69634, 69733, 69880, 69922, 69933, 70063, 70090, 70101, 70425, 70518, 70593, 70680, 70949, 71009, 71086, 71154, 71196, 71298, 71338, 71447, 71488, 71554, 71734, 71773, 71813, 71877, 71948, 71950, 72063, 72169, 72219, 72289, 72384, 72484, 72514, 72585, 72615, 72642, 72661, 72700, 72777, 72876, 72925, 72953, 73381, 73396, 73416, 73458, 73700, 73874, 73907, 73953, 74045, 74165, 74213, 74707, 74749, 75012, 75180, 75281, 75366, 75505, 75603, 75647, 75685, 76050, 76268, 76348, 76439, 76530, 76551, 76888, 77075, 77119, 77184, 77199, 77334, 77397, 77725, 77774, 77815, 77946, 78102, 78176, 78518, 78590, 78661, 78823, 78920, 79043, 79057, 79178, 79260, 79446, 79476, 79496, 79558, 79659, 79740, 79875, 79940, 80256, 80399, 80442, 80469, 80529, 80585, 80675, 80713, 80764, 80859, 80981, 81052, 81081, 81205, 81324, 81388, 81428, 81524, 81575, 81587, 82195, 82204, 82249, 82263, 82656, 82859, 82869, 82883, 82954, 82990, 83060, 83082, 83093, 83136, 83236, 83268, 83598, 83717, 83738, 83768, 83803, 83954, 84015, 84018, 84056, 84170, 84214, 84228, 84940, 84960, 85094, 85144, 85662, 85750, 85893, 85900, 86008, 86023, 86060, 86101, 86138, 86379, 86403, 86455, 86588, 86610, 86654, 86686, 86835, 87005, 87086, 87212, 87302, 87380, 87459, 87547, 88380, 88686, 89128, 89206, 89297, 89485, 89519, 89672, 89869, 90822, 91110, 91186, 91955, 92034, 92065, 92166, 92181, 92232, 92265, 92372, 92459, 92634, 92704, 92888, 93284, 93615, 93708, 93748, 93775, 93866, 93984, 94002, 94037, 94256, 94694, 94717, 94946, 95001, 95248, 95386, 95388, 96025, 96579, 97113, 97862, 98424, 98815, 98942, 99064, 99170, 99414, 99526, 99985, 100420, 100464, 100753, 100773, 100798, 100846, 100911, 100979, 101034, 102075, 102173, 102245, 102390, 102431, 102644, 102789, 103127, 103406, 103662, 103679, 103750, 105015, 105319, 105532, 105905, 106102, 106335, 106384, 106500, 106609, 106624, 106851, 106985, 107274, 107405, 107705, 107747, 107829, 107879, 107927, 108105, 108618, 108622, 108931, 108980, 109611, 110086, 110858, 110868, 110982, 111263, 111383, 111530, 111959, 112215, 112370, 112596, 112866, 112962, 113019, 113134, 113158, 113203, 113312, 113372, 113535, 113690, 113887, 113907, 113968, 114042, 114066, 114169, 114208, 114345, 114401, 114448, 114578, 114587, 114595, 115008, 115383, 115503, 115870, 116574, 116585, 116620, 118698, 118712, 118785, 118899, 118921, 118989, 119009, 119088, 119115, 119166, 119239, 119444, 119479, 119542, 119624, 119655, 119702, 119723, 119789, 119823, 119851, 119971, 120046, 121222, 121343, 121689, 122946, 123138, 123197, 123224, 123565, 123628, 123710, 123802, 123812, 123821, 123871, 124055, 124349, 124399, 124432, 124571, 124596, 124623, 124690, 125006, 125850, 125875, 125980, 126048, 126144, 126201, 126262, 126320, 126347, 126688, 126751, 126833, 126925, 126935, 126944, 126994, 127178, 127472, 127522, 127555, 127694, 127719, 127746, 127813, 128129, 128973, 128998, 129103, 129171, 129267, 129324, 129385, 133481, 133527, 133578, 133618, 133644, 133723, 133791, 133916, 134240, 134377, 134407, 135471, 136258, 136643, 137645, 137827, 137877, 137896, 138700, 138738, 139817, 139924, 139944, 139984, 140005, 140151, 140166, 140335, 140589, 140780, 141585, 141604, 141888, 141911, 142288, 142832, 144486, 144553, 144601, 147110, 147261, 147680, 148248, 148382, 148541, 148605, 148612, 148719, 148740, 148851, 148944, 149013, 149381, 149704, 149798, 149842, 149942, 150064, 150083, 150306, 150387, 150451, 150477, 150546, 150644, 150845, 150905, 151020, 151116, 151303, 151410, 151471, 151642, 151693, 151733, 151758, 151900, 152025, 152105, 152373, 152573, 152706, 152769, 152837, 152877, 152889, 153239, 153301, 153420, 153436, 153472, 153556, 153590, 153605, 153670, 153743, 153894, 153917, 154042, 154293, 154479, 154699, 154838, 155067, 155100, 155337, 155469, 155599, 155688, 155762, 155800, 155856, 155889, 155910, 156254, 156424, 156481, 156527, 156654, 156678, 156715, 156730, 156779, 156826, 156851, 156868, 156945, 157010, 157017, 157086, 157100, 157173, 157426, 157625, 157698, 157770, 158234, 158261, 158275, 158282, 158299, 158307, 158328, 158337, 158355, 158375, 158404, 158414, 158430, 158471, 158510, 158560, 158585, 158635, 158815, 158828, 158915, 158940, 158993, 158996, 159036, 159124, 159231, 159273, 159314, 159335, 159362, 159381, 159389, 159429, 159433, 159535, 159543, 159574, 159606, 159614, 159626, 159635, 159679, 159690, 159694, 159719, 159731, 159772, 159790, 159910, 160194, 160318, 160393, 160459, 160541, 160548, 160577, 160721, 160929, 160985, 161076, 161098, 161210, 161378, 161529, 161625, 161866, 161909, 161940, 161997, 162099, 162183, 162232, 162398, 162502, 162556, 162601, 162664, 162764, 162951, 162971, 162982, 163145, 163334, 164099, 164212, 164268, 164293, 164443, 164469, 164506, 164604, 164820, 164840, 164872, 164914, 164944, 164977, 164983, 165002, 165157, 165184, 165209, 165274, 165296, 165336, 165437, 165451, 165462, 165508, 165556, 165650, 165735, 165791, 165814, 165820, 165879, 165882, 165899, 165957, 165976, 166032, 166074, 166089, 166119, 166247, 166263, 166297, 166335, 166376, 166416, 166440, 166446, 166461, 166502, 166518, 166521, 166563, 166588, 166683, 166700, 166852, 166873, 166957, 166976, 167079, 167196, 167209, 167224, 167240, 167287, 167300, 167347, 167434, 167460, 167497, 167560, 167646, 167677, 167760, 167806, 167876, 167973, 168034, 168071, 168309, 168387, 168445, 168463, 168475, 168481, 168514, 168548, 168564, 168633, 168714, 168737, 168771, 168833, 168849, 168912, 169089, 169176, 169305, 169384, 169402, 169481, 169510, 169530, 169590, 169675, 169685, 169752, 169771, 169817, 169842, 169873, 170207, 171535, 171609, 171865, 171884, 172021, 172321, 172893, 172973, 173027, 173094, 173105, 173162, 173165, 173177, 173219, 173234, 173257, 173282, 173356, 173411, 173433, 173504, 173514, 173916, 173930, 173947, 173960, 174027, 174045, 174070, 174072, 174135, 174178, 174261, 174296, 174428, 174460, 174472, 174525, 174541, 174693, 174711, 174717, 174747, 174769, 174852, 175106, 175139, 175179, 175226, 175305, 175363, 175388, 175406, 175645, 175654, 175657, 175688, 175725, 175851, 175877, 175890, 175904, 175908, 176002, 176051, 176054, 176137, 176172, 176230, 176397, 176420, 176422, 176481, 176528, 176542, 176547, 176580, 176588, 176606, 176622, 176631, 176671, 176804, 177073, 177146, 177282, 177315, 177422, 177458, 177700, 177889, 177920, 177963, 177976, 178007, 178219, 178458, 178491, 178558, 178678, 178938, 178992, 179265, 179456, 179530, 179928, 180016, 180157, 180170, 180204, 180249, 180295, 180740, 180856, 180892, 180963, 180991, 181014, 181055, 181087, 181214, 181311, 181516, 181542, 181603, 181667, 181844, 181893, 181990, 182012, 182223, 182279, 182317, 182345, 182476, 182591, 183149, 183366, 183564, 183598, 183775, 183819, 184086, 184151, 184209, 184251, 184292, 184307, 184418, 184478, 184535, 184555, 184582, 184603, 184765, 184771, 184827, 184923, 185351, 185432, 185456, 185554, 185559, 185660, 185697, 185700, 185792, 185855, 185945, 185993, 186354, 186369, 186385, 186608, 186806, 186986, 187076, 187184, 187321, 187382, 187437, 187573, 187631, 187753, 187803, 187952, 188051, 188119, 188245, 188340, 188415, 188434, 188496, 188510, 188539, 188561, 188582, 188609, 188620, 188653, 188681, 188691, 188703, 188717, 188743, 188816, 188825, 188840, 188855, 188972, 189032, 189050, 189105, 189115, 189149, 189157, 189184, 189206, 189302, 189307, 189379, 189405, 189413, 189509, 189539, 189542, 189561, 189574, 189591, 189616, 189653, 189675, 189695, 189698, 189714, 189741, 189768, 189778, 189786, 189808, 189814, 189823, 189887, 189936, 189978, 190081, 190124, 190169, 190195, 190230, 190268, 190335, 190493, 190597, 190709, 190752, 190781, 190988, 191186, 191222, 191357, 191532, 191614, 191785, 191863, 191904, 192091, 192129, 192171, 192297, 192470, 192483, 192572, 192626, 192715, 192914, 192928, 192979, 193304, 193329, 193559, 193693, 193746, 193786, 193883, 194016, 194102, 194246]


hidalgo  esta en los indices [4336, 5444, 17316, 37834, 38090, 38140, 38177, 38277, 38690, 39208, 39552, 40280, 40605, 40675, 40794, 41030, 41058, 41304, 42228, 64003, 74776, 80255, 83208, 83418, 83488, 83541, 83628, 83638, 83673, 91717, 91732, 115080, 124897, 128020, 132112, 139150, 169383, 184530, 193831, 193899]

merced  esta en los indices [761, 820, 918, 1562, 1978, 2219, 2284, 2371, 3016, 3125, 3348, 3654, 4570, 4981, 5039, 5057, 5167, 5196, 5373, 5439, 5612, 6064, 6272, 6304, 6339, 6393, 6433, 6798, 7392, 8322, 9703, 9714, 10037, 12766, 12957, 13399, 13691, 15105, 15221, 15249, 15416, 15481, 15593, 15691, 15771, 15875, 15979, 16052, 16373, 16633, 16637, 16736, 16774, 16792, 17603, 17655, 17824, 18736, 19074, 19094, 20141, 20264, 20372, 20515, 20541, 20618, 20951, 21071, 21117, 21126, 21777, 21981, 22706, 22743, 22803, 23072, 23137, 24312, 24654, 24943, 25010, 25151, 25174, 25183, 25628, 26142, 26225, 26265, 26523, 26950, 27362, 27390, 27444, 28146, 28209, 29089, 29417, 29494, 29513, 29684, 29896, 29919, 30018, 30366, 30619, 30673, 31233, 31923, 33053, 33105, 33224, 34075, 34109, 34652, 34727, 35592, 35872, 35952, 36466, 36896, 36908, 37003, 37221, 37251, 37624, 37642, 37744, 38156, 39211, 39298, 40134, 40164, 40198, 40501, 40641, 40773, 40792, 40874, 41075, 41976, 42034, 42498, 42761, 42798, 43235, 43290, 43326, 43351, 43381, 43856, 43967, 44103, 44117, 44125, 44224, 44277, 44361, 44684, 44853, 45231, 45246, 45272, 45293, 45811, 45851, 46096, 46111, 46235, 46643, 46661, 47975, 48009, 48686, 49399, 49417, 50731, 50765, 51442, 52466, 52513, 53017, 54464, 54507, 54877, 54912, 58048, 58659, 59118, 59148, 59187, 59793, 60812, 62139, 62198, 62235, 62289, 62384, 62502, 62606, 62628, 62713, 63013, 63040, 63090, 63123, 63316, 63323, 63344, 63369, 63412, 63584, 63694, 63779, 64224, 64240, 64310, 64443, 64458, 65245, 65404, 66312, 67646, 68054, 69194, 69210, 69227, 69285, 69413, 69445, 69496, 69590, 71005, 71296, 71788, 72094, 75725, 75824, 75919, 76016, 76028, 76175, 76189, 76255, 76382, 76411, 76486, 76510, 76941, 76965, 77616, 77673, 78094, 78149, 78300, 78510, 78637, 80088, 80730, 81056, 81703, 81721, 81881, 83011, 83040, 84267, 84318, 84402, 84752, 84971, 85096, 85139, 87037, 87061, 87259, 87295, 87390, 87515, 88232, 89030, 89197, 89967, 91017, 91623, 91882, 94034, 95878, 96639, 97552, 98301, 99151, 100011, 100037, 101168, 101175, 102345, 105804, 106250, 106281, 106295, 106312, 106327, 106474, 106607, 108453, 109867, 111854, 112307, 112576, 112740, 112811, 112819, 113514, 114104, 114231, 114420, 115707, 116761, 116932, 117252, 117299, 118350, 118381, 118987, 119003, 119012, 119994, 120651, 120666, 121515, 121666, 122710, 122713, 122864, 123819, 123835, 123860, 123865, 123877, 123893, 123914, 123964, 124021, 124386, 124627, 124676, 124695, 124797, 124992, 125004, 125646, 125669, 125688, 125769, 125873, 125943, 126942, 126958, 126983, 126988, 127000, 127016, 127037, 127087, 127144, 127509, 127750, 127799, 127818, 127920, 128115, 128127, 128769, 128792, 128811, 128892, 128996, 129066, 129667, 130069, 130095, 130113, 130131, 130280, 130299, 130329, 130517, 130690, 130706, 130841, 131009, 131043, 131302, 131339, 131456, 131520, 132561, 133275, 134026, 134224, 134284, 134302, 134351, 134426, 135725, 136206, 136282, 136462, 136592, 137001, 137222, 137251, 137261, 137882, 138783, 138878, 139162, 139228, 139283, 139298, 139325, 139337, 139369, 139408, 139423, 139472, 139503, 140310, 140323, 140470, 140492, 141266, 141544, 142867, 143401, 144007, 144035, 144062, 144232, 144274, 147493, 148710, 148847, 148985, 153105, 153351, 153356, 154247, 154604, 154643, 154686, 155589, 156075, 157323, 157397, 157556, 157938, 158280, 158335, 160256, 160299, 160430, 163207, 163262, 163288, 164981, 165920, 165942, 165974, 166004, 166038, 166229, 166500, 166740, 166796, 168791, 168805, 168829, 168856, 168875, 168918, 169112, 169171, 169240, 169262, 169301, 173226, 173308, 174143, 175441, 175533, 175563, 175846, 177048, 177165, 177190, 177943, 177995, 178082, 178153, 178666, 178898, 179984, 180009, 180525, 180788, 180820, 181277, 184064, 184131, 185795, 185829, 185859, 185892, 186763, 186789, 186959, 187056, 187541, 187719, 188494, 188616, 188636, 188658, 188698, 188814, 188829, 188894, 188926, 189181, 189335, 189358, 189396, 189461, 189482, 189586, 190354, 190392, 190553, 191631, 191678, 192278, 192585, 192608, 193152, 193263]
